apps/server/src/modules/authentication/strategy/local.strategy.ts

```diff
@@ -0,0 +1,37 @@
+import { Strategy } from 'passport-local';
+import { PassportStrategy } from '@nestjs/passport';
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import bcrypt from 'bcryptjs';
+import { UserRepo } from '@shared/repo';
+import { ICurrentUser } from '@shared/domain';
+import { CurrentUserMapper } from '@shared/domain/mapper/current-user.mapper';
+import { AuthenticationService } from '../services/authentication.service';
+
+@Injectable()
+export class LocalStrategy extends PassportStrategy(Strategy) {
+	constructor(private readonly authenticationService: AuthenticationService, private readonly userRepo: UserRepo) {
+		super();
+	}
+
+	async validate(username?: string, password?: string): Promise<ICurrentUser> {
+		if (!username || !password) {
```

Comment:
Also the same, use private checkXX methodes for each if throw block.

Reply:
Thease are type guards. Putting type gurads into a separate function doesn't bring any additional value.